### PR TITLE
Get-DbaDbBackupHistory - change AgCheck to SuppressAgCheck and use it in Backup-DbaDatabase

### PR DESCRIPTION
**Pull request by andreasjordan**
This change is not a classic bug fix, but it will help in situations described in #6604 and generally speed up Backup-DbaDatabase because in the call to get the record for the just taken backup it does not ask other instances.

I changed the name of a parameter to better reflect its purpose. And it makes this parameter a public parameter and not only an internal one. The name "SuppressAgCheck" is only an idea, it could also be "NoAgCheck" or some other name. But I think "AgCheck" is wrong because this is positiv, so I assume some kind of active check related to AGs - but the parameter itself is negative and suppresses some action. But I added an alias for backword compatibility, just in case someone uses this (internal) parameter.

I already mentioned this branch here: https://github.com/sqlcollaborative/dbatools/issues/6604#issuecomment-672638390 and hoped for a discussion on it. So with this pull request I want to get feedback on my ideas so that my ideas are not quietly lost.

I also fixed some case issues that VSCode reported.

### CODE REVIEW

**Comment by andreasjordan** [comment]
I added an example and would be happy if a native english speeker could have a look and rephrase it.

**Comment by wsmelton** [comment]
The discussion we've had was to actually completely remove it. Don't believe that we every fully decided but there has been confusion around it and just adding a new parameter name I don't think will help.

If folks want to get backup history around an AG it deserves a dedicated command. I think that will make it more clear.

We use exclude parameters to filter objects in our module. If this was a test command then I'd see use for a parameter to skip some check or functionality but not with a Get function.

It will offer easier adoption and allow I think for separation of duties (so to speak) to just make a new command. Something like `Get-DbaAgBackupHistory`. We can also have better control of how it accepts instances with non-default ports potentially as well.

Just my money's worth...

**Comment by andreasjordan** [comment]
I would vote for removing the AG-check-code from `Get-DbaDbBackupHistory`. That would speed up `Backup-DbaDatabase` as well. And although it is documented, checking other instances than the one you submit in the command is a bit "surprising".

I would also vote for a new command `Get-DbaAgBackupHistory`. There we could have a similar parameter set like in `Add-DbaAgDatabase` with an optional parameter `-Secondary` for those with custom ports. And we could have `-AvailabilityGroup` to get the history for all databases in that group.

The code would become more readable because we have no recursion anymore.

Should I try to create the new command - as part of my training in developing dbatools?

Just mention @potatoqualitee and @Stuart-Moore as authors of the command here.

**Comment by andreasjordan** [comment]
@wsmelton Do you remember where the discussion took place and who was involved? I would like to push this issue a little bit...

**Comment by wsmelton** [comment]
Slack with Stuart.

File: functions/Get-DbaDbBackupHistory.ps1 [hunk 1]
```diff
@@ -173,11 +179,12 @@ function Get-DbaDbBackupHistory {
         [switch]$LastLog,
         [string[]]$DeviceType,
         [switch]$Raw,
-        [bigint]$LastLsn,
```
Comment (Stuart-Moore) [inline functions/Get-DbaDbBackupHistory.ps1:176]: This stays, don't remove it

File: functions/Get-DbaDbBackupHistory.ps1 [hunk 2]
```diff
@@ -173,11 +179,12 @@ function Get-DbaDbBackupHistory {
...
-        [switch]$IncludeMirror,
```
Comment (Stuart-Moore) [inline functions/Get-DbaDbBackupHistory.ps1:177]: Dont remove this please

**Comment by Stuart-Moore** [comment]
Rip it out

You're going to end up with some recursion in the new cmd  as you're  going to have to call Get-DbaDbBackuphistory across replicas anyway to make sure it's using the same filtering logic and output types.

Please don't try to rewrite that as we've just about got agreement on these methods after $DEITY knows how long :) and the wrong output type will bork the restore commands when using `TrustBackupHistory` options

ignore the review bits, was confused as you moved some stuff around and I'm suffering on bad hotel wifi

**Comment by andreasjordan** [comment]
Started a brand new pull request: #6799 so closing this one.

**Comment by wsmelton** [comment]
Are we not going to rip out the parameter and associated code?

@Stuart-Moore unless I'm misreading your post...

**Comment by andreasjordan** [comment]
I will certainly continue to work on it and rip out the parameter and assiciated code. I just would like to have a confirmation that the new code does exactly what the old code did.
Or is it enough to be able to return to this commit point later? Then I will probably work on that today.

**Pull request closed without merging**